test/Generics/protocol_requirement_signatures.swift

```diff
@@ -0,0 +1,60 @@
+// RUN: %target-typecheck-verify-swift -typecheck %s -verify
+// RUN: %target-typecheck-verify-swift -typecheck -debug-generic-signatures %s > %t.dump 2>&1 
+// RUN: %FileCheck %s < %t.dump
+
+// CHECK-LABEL: .P1@
+// CHECK-NEXT: Requirements:
+// CHECK-NEXT:   Self : P1 [protocol_self]
+// CHECK-NEXT: Generic signature: <Self where Self : P1>
+// CHECK-NEXT: Canonical generic signature: <τ_0_0 where τ_0_0 : P1>
```

Comment:
Why not? Do you mean we should treat `ProtocolRequirementSignatureSelf` like they're redundant? I recall from earlier discussions that we'd just drop this requirement when doing later processing, but it would also make sense to just not include it in the first place.